src/crates-io/lib.rs

```diff
@@ -36,6 +36,7 @@ pub enum Error {
     Unauthorized,
     TokenMissing,
     Io(io::Error),
+    CrateNotFound,
```

Comment:
This may be best renamed just `NotFound` because a 404 doesn't always mean it's reference a crate, just in the context of some APIs